src/TCommandLine.cpp

```diff
@@ -279,192 +212,227 @@ bool TCommandLine::event( QEvent * event )
+                    mUserKeptOnTyping = true;
+                    mAutoCompletionCount = -1;
+                } else {
+                    mUserKeptOnTyping = false;
                 }
+                spellCheck();
+                return false;
+            }
         }
-
     }
 
-    return QPlainTextEdit::event( event );
+    return QPlainTextEdit::event(event);
 }
 
-void TCommandLine::focusInEvent( QFocusEvent * event )
+void TCommandLine::focusInEvent(QFocusEvent* event)
 {
     textCursor().movePosition(QTextCursor::Start);
-    textCursor().movePosition(QTextCursor::Right, QTextCursor::KeepAnchor, mSelectedText.length() );
+    textCursor().movePosition(QTextCursor::Right, QTextCursor::KeepAnchor, mSelectedText.length());
 
     mpConsole->console->forceUpdate();
     mpConsole->console2->forceUpdate();
-    QPlainTextEdit::focusInEvent( event );
+    QPlainTextEdit::focusInEvent(event);
 }
 
-void TCommandLine::focusOutEvent( QFocusEvent * event )
+void TCommandLine::focusOutEvent(QFocusEvent* event)
 {
-    if( textCursor().hasSelection() )
-    {
+    if (textCursor().hasSelection()) {
         mSelectionStart = textCursor().selectionStart();
         mSelectedText = textCursor().selectedText();
-    }
-    else
-    {
+    } else {
         mSelectionStart = 0;
         mSelectedText = "";
     }
-    QPlainTextEdit::focusOutEvent( event );
+    QPlainTextEdit::focusOutEvent(event);
 }
 
 void TCommandLine::adjustHeight()
 {
     int lines = document()->size().height();
-    int fontH = QFontMetrics( mpHost->mDisplayFont ).height();
-    if( lines < 1 ) lines = 1;
-    if( lines > 10 ) lines = 10;
+    int fontH = QFontMetrics(mpHost->mDisplayFont).height();
+    if (lines < 1) {
+        lines = 1;
+    }
+    if (lines > 10) {
+        lines = 10;
+    }
     int _baseHeight = fontH * lines;
     int _height = _baseHeight + fontH;
-    if( _height < mpHost->commandLineMinimumHeight )
+    if (_height < mpHost->commandLineMinimumHeight)
         _height = mpHost->commandLineMinimumHeight;
-    if( _height > height() || _height < height() )
-    {
-        mpConsole->layerCommandLine->setMinimumHeight( _height );
-        mpConsole->layerCommandLine->setMaximumHeight( _height );
+    if (_height > height() || _height < height()) {
+        mpConsole->layerCommandLine->setMinimumHeight(_height);
+        mpConsole->layerCommandLine->setMaximumHeight(_height);
         int x = mpConsole->width();
         int y = mpConsole->height();
-        QSize s = QSize(x,y);
+        QSize s = QSize(x, y);
         QResizeEvent event(s, s);
-        QApplication::sendEvent( mpConsole, &event);
+        QApplication::sendEvent(mpConsole, &event);
     }
 }
 
 void TCommandLine::spellCheck()
 {
-    if( ! mpHost->mEnableSpellCheck ) return;
+    if (!mpHost->mEnableSpellCheck) {
+        return;
+    }
 
     QTextCursor oldCursor = textCursor();
     QTextCharFormat f;
-    QColor cred = QColor(255,0,0);
+    QColor cred = QColor(255, 0, 0);
```

Comment:
Copy constructors have to be present for copy initialization to be conforming, but they are allowed to be elided.